tests/testthat/test-unique.R

```diff
@@ -0,0 +1,22 @@
+test_that("includes implicit missing values", {
+  x <- factor(c(NA, "x"))
+  expect_equal(fct_unique(x), fct(c("x", NA)))
+})
+
+test_that("preserve position of explicit NA levels", {
+  x <- fct(c(NA, "x"), levels = c(NA, "x"))
+  expect_equal(fct_unique(x), x)
+})
+
+test_that("preserve mix of implicit and explicit NA", {
```

Comment:
```suggestion
test_that("preserve mix of implicit and explicit NA (#328)", {
```

For future readers, we discovered that while most factors replace `NA` with an integer code when `NA` is used as an explicit factor level, it is actually documented in `?factor` that `is.na<-.factor` can be used to simultaneously have `NA` as a code and `NA` as a completely separate explicit factor level that maps to an integer code.

``` r
x <- factor(c("a", NA, NA, "b"), levels = c("a", "b", NA), exclude = NULL)
unclass(x)
#> [1] 1 3 3 2
#> attr(,"levels")
#> [1] "a" "b" NA

is.na(x)[3] <- TRUE

x
#> [1] a    <NA> <NA> b   
#> Levels: a b <NA>
unclass(x) # technically 3 and NA are different
#> [1]  1  3 NA  2
#> attr(,"levels")
#> [1] "a" "b" NA
```

So this test ensures we support that

Reply:
Pulling this comment out to the top-level so it doesn't get lost.